sklearn/tree/_tree.pyx

```diff
@@ -375,9 +434,20 @@ cdef class BestFirstTreeBuilder(TreeBuilder):
 
         with nogil:
             # add root to frontier
-            rc = self._add_split_node(splitter, tree, 0, n_node_samples,
-                                      INFINITY, IS_FIRST, IS_LEFT, NULL, 0,
-                                      &split_node_left)
+            rc = self._add_split_node(
+                                      splitter,         # splitter
+                                      tree,             # tree
+                                      0,                # start
+                                      n_node_samples,   # end
+                                      INFINITY,         # impurity
+                                      IS_FIRST,         # is_first
+                                      IS_LEFT,          # is_left
+                                      NULL,             # parent
+                                      0,                # depth
+                                      -INFINITY,        # lower_bound
+                                      INFINITY,         # upper_bound
+                                      &split_node_left  # res
```

Comment:
Potentially using key-words arguments reads better (I should have had proposed it in the first place!).
```suggestion
                splitter=splitter,
                tree=tree,
                start=0,
                end=n_node_samples,
                impurity=INFINITY,
                is_first=IS_FIRST,
                is_left=IS_LEFT,
                parent=NULL,
                depth=0,
                lower_bound=-INFINITY,
                upper_bound=INFINITY,
                res=&split_node_left,
```

Key-words arguments can also be used in other places.

Reply:
Uh oh, this didn't go well...